src/main/java/com/google/cloud/tools/appengine/operations/Deployment.java

```diff
@@ -146,4 +156,20 @@ void deployConfig(String filename, DeployProjectConfigurationConfiguration confi
       throw new AppEngineException(ex);
     }
   }
+
+  @VisibleForTesting
+  @Nullable
+  String processMode(@Nullable String modeFromConfig) throws AppEngineException {
+    if (modeFromConfig == null) {
+      return null;
+    }
+    String trimmedModeFromConfig = modeFromConfig.trim();
+    Optional<String> matchingMode =
+        GCLOUD_MODES
+            .stream()
+            .filter(mode -> mode.equalsIgnoreCase(trimmedModeFromConfig))
```

Comment:
nit: `.filter(trimmedModeFromConfig::equalsIgnoreCase)` is more compact

I wouldn't care about supporting upper/mixed cases though, but this doesn't hurt.